packages/flutter/lib/src/material/material_localizations.dart

```diff
@@ -48,3 +47,45 @@ class MaterialLocalizations {
     return Localizations.of<MaterialLocalizations>(context, MaterialLocalizations);
   }
 }
+
+/// Localized strings for the material widgets.
+class DefaultMaterialLocalizations implements MaterialLocalizations {
+  /// Construct an object that defines the material widgets' localized strings
+  /// for the given `locale`.
+  const DefaultMaterialLocalizations(this.locale) : assert(locale != null);
+
+  /// The locale for which the values of this class's localized resources
+  /// have been translated.
+  final Locale locale;
+
+  String _lookup(String name) {
+    final Map<String, String> nameToValue = localizations[locale.toString()]
+      ?? localizations[locale.languageCode]
+      ?? localizations['en'];
```

Comment:
we shouldn't need to do this here, the load method should just map from the given Locale to the nearest matching supported locale and return the object appropriately configured. That way we don't have to do this dance every build, for instance.